invenio_communities/views.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2019 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Blueprint definitions."""
+
+from __future__ import absolute_import, print_function
+
+from flask.views import MethodView
+from webargs.flaskparser import use_kwargs
+from webargs import fields, validate
+from invenio_db import db
+
+from flask import Blueprint
+from invenio_communities.api import CommunityMembersCls, MembershipRequestCls
+from flask_security import current_user
+
+def create_blueprint_from_app(app):
+    community_members_rest_blueprint = Blueprint(
+        'invenio_communities',
+        __name__,
+        url_prefix="",
+        template_folder="./templates"
+    )
+    comm_view = CommunityMembersResource.as_view(
+        'community_members_api'
+
+    )
+    community_members_rest_blueprint.add_url_rule(
+        '/communities/<{0}:pid_value>/members'.format(
+            'pid(comid,record_class="invenio_communities.api:Community",'
+            'object_type="com")'),
+        view_func=comm_view,
+    )
+
+    request_view = MembershipRequest.as_view(
+        'community_requests_api'
+    )
+    community_members_rest_blueprint.add_url_rule(
+        '/communities/requests/<token>',
```

Comment:
Since this is the REST API, it should follow the resources model, i.e. expose:

-  `(GET|POST) /api/communities/requests/`
- `(GET|PUT|DELETE) /api/communities/requests/<request-id>`

The UI view `/communities/requests?token=<...>` is responsible for parsing/validating the token, and rendering the appropriate accept/reject box with the resolved `request-id` injected in the Jinja (to be used by the Javascript app to make a `POST /api/communities/requests/<request-id>` request). Let's discuss IRL though, because there are some tricky details on how auth would work